packages/roosterjs-content-model/lib/modelApi/selection/iterateSelections.ts

```diff
@@ -103,9 +105,63 @@ export function iterateSelections(
                 });
 
                 if (segments.length > 0) {
+                    if (segments.length == 1 && segments[0].segmentType == 'SelectionMarker') {
+                        let markerSelectionIndex = block.segments.indexOf(segments[0]);
+                        for (let i = markerSelectionIndex - 1; i >= 0; i--) {
+                            if (block.segments[i].segmentType == 'Text') {
+                                const found = findDelimiter(
+                                    block.segments[i] as ContentModelText,
+                                    false
+                                );
+                                if (found != null) {
+                                    let blenght = block.segments[i] as ContentModelText;
+                                    if (found == blenght.text.length) {
+                                        break;
+                                    }
+                                    splitTextSegment(
+                                        block.segments as ContentModelText[],
+                                        i,
+                                        found
+                                    );
+                                    //block.segments[i + 1].isSelected = true;
+                                    segments.push(block.segments[i + 1]);
+                                    break;
+                                } else {
+                                    segments.push(block.segments[i]);
+                                }
+                            } else {
+                                break;
+                            }
+                        }
+                        markerSelectionIndex = block.segments.indexOf(segments[0]);
+                        for (let i = markerSelectionIndex + 1; i < block.segments.length; i++) {
+                            if (block.segments[i].segmentType == 'Text') {
+                                const found = findDelimiter(
+                                    block.segments[i] as ContentModelText,
```

Comment:
I feel I am doing a lot of casting to ContentModelText

Reply:
to avoid type casting, you can do:
```ts
const segment = block.segments[i];

if (segment.segmentType == 'Text') {
   // Now you can use segment as if it is a Text without any type casting because you have already check its type. This is supported by typescript
}
```